cpp/src/parquet/encoding.cc

```diff
@@ -442,12 +442,12 @@ class DictEncoderImpl : public EncoderImpl, virtual public DictEncoder<DType> {
   constexpr static int32_t kDataPageBitWidthBytes = 1;
 
   explicit DictEncoderImpl(const ColumnDescriptor* desc, MemoryPool* pool)
-      : EncoderImpl(desc, Encoding::PLAIN_DICTIONARY, pool),
+      : EncoderImpl(desc, Encoding::RLE_DICTIONARY, pool),
```

Comment:
I'm not sure if this is a minor breaking change to those legacy readers which stick to `PLAIN_DICTIONARY`. cc @jorisvandenbossche

Reply:
IMO this is only a Parquet internal API, maybe user shouldn't depend on this?

Reply:
It doesn't change what we written to storage? The storage would final decided by `Properties::dictionary_index_encoding()`

Reply:
Yes, you're right. I forgot this.